app/services/submission_service.py

```diff
@@ -40,6 +47,14 @@ async def create_submission(
         self.session.add(submission)
         self.session.commit()
 
+        student_notebook_upload = await grading_service.get_student_notebook_upload(submission, student_notebook_content.encode())
```

Comment:
I'm worried that if there's an error thrown in the stuff you've added the stuff done previously needs to be rolled back. Maybe move this part to below the code you added?

```
self.session.add(submission)
self.session.commit()
```

Reply:
this change is done in submission downsync. I'm not particularly sure how the syncing change above got included in manual-grading and submission-downsync but not in submission-upload-cleanup which is the parent of both...

Reply:
So...what needs to happen here? Can we make this code exactly the way it should be for now?

Reply:
This code is outdated

Reply:
In fairness, even after merging in the initial submission sync, the actual cleanup code seems to live on the second "submission downsync" PR, so the issue still persists it's just been relocated.